### PR TITLE
Improve table header styling

### LINKED ISSUE #1077: Table column headers are misaligned
### Describe the bug

There are a number of cosmetic issues with the table column headers,

- When a table is created the column headers are shifted slightly to the right. They are not correctly aligned with the table columns.
- When the first column header is clicked to enable sorting the header is expanded to show the direction of sort graphic, this in turn expands the whole column horizontally which is annoying. It would ideal to pre-allocate space for the sorting icon so the column header and its text doesn't jump about.

### How to reproduce

Create a simple table and click on the first column header.

### Example code

_No response_

### Relevant output

_No response_

### Platform

macOS

### PR DESCRIPTION
**Pull request by kkoreilly**
Fixes #1077.

### CODE REVIEW

File: core/toolbar.go [hunk 1]
```diff
@@ -217,10 +217,10 @@ func ToolbarStyles(w Widget) {
 	wb.FinalStyler(func(s *styles.Style) {
 		if s.Direction == styles.Row {
 			s.Grow.Set(1, 0)
-			s.Padding.SetHorizontal(units.Dp(ConstantSpacing(16)))
+			s.Padding.SetHorizontal(units.Dp(16))
```
Comment (rcoreilly) [inline core/toolbar.go:220]: why is this in here and what is the motivation?

**Review by rcoreilly — COMMENTED** [review]
can you also implement in the tensor tableview so I can see what it looks like at scale, on the plotview data?

also what is up with the test?  just that random thing?

File: core/toolbar.go (hunk 1 above)
Comment (rcoreilly) [reply core/toolbar.go:220]: If this is to improve alignment in the table, which it seems like from your comment, then it should be styled specifically in the table, not in the generic toolbar styles.  For those using compact spacing, it is essential to have this constant spacing for the general toolbar case.

**Comment by kkoreilly** [comment]
Yes, I just updated `tensorcore.Table` to the new changes, and it looks much better to me on the plot example data. The test failure is one of our two random test failures; I will work on making that more robust now.

File: core/toolbar.go (hunk 1 above)
Comment (kkoreilly) [reply core/toolbar.go:220]: With our new border radius based parent border radius encroachment fix (#660), we don't need constant spacing on the toolbar anymore.

Comment (kkoreilly) [reply core/toolbar.go:220]: The table *does* now have specific styling, which is zero padding, but again, independent of that the toolbar does not need to have the constant spacing anymore.

Comment (rcoreilly) [reply core/toolbar.go:220]: ok, I tested it on various use-cases and toolbars look fine..

Comment (kkoreilly) [reply core/toolbar.go:220]: So did I.

**Comment by rcoreilly** [comment]
FYI the change results in 3.5 fewer columns of data visible in the plot data table.  also it seems like it is reserving too much extra space?  e.g., the Epoch column doesn't need that much?

New:

image
 
Existing:

image

**Comment by rcoreilly** [comment]
Indicator styling definitely looks better though!

**Comment by kkoreilly** [comment]
You are certainly free to investigate the space allocation!

**Comment by kkoreilly** [comment]
Also @rcoreilly do you have any interest in fixing this test timeout issue? You can look at the CI logs.

**Comment by rcoreilly** [comment]
no interest in that.  should I merge anyway?

**Review by rcoreilly — APPROVED** [review]
Filed #1080 to deal with optimizing spacing, later..

**Comment by kkoreilly** [comment]
Yes, merge, since CI shouldn't fail on main. We should definitely fix this though.

**Pull request merged**